### ISSUE TITLE
TransformToExternalPath is necessary when using SwaggerRoutes with OWIN

### ISSUE THREAD

**Issue by ideafixxxer** [RicoSuter/NSwag#2115]
Related: #1914 

When using multiple Swagger Documents with UI3 on ASP.NET OWIN, the default implementation of TransformToExternalPath results in all references to swagger.json to be unresolved.

For example:
``` C#
settings.MiddlewareBasePath = "/swagger";
settings.SwaggerRoutes.Add(new SwaggerUi3Route("v1", "/swagger/v1/swagger.json"));
```
My app is hosted at `https://localhost/services/api/`, however the web browser is searching the file at `https://localhost/v1/swagger.json`. Adding the following line fixes the issue:
``` C#
settings.TransformToExternalPath = (url, request) => request.PathBase + url;
```
My strong opinion is that it should the default implementation.

**Comment by RicoSuter** [comment]
> My strong opinion is that it should the default implementation.

I think you are right with this... I'm a bit hesitant as this might be a breaking change...

**Comment by ideafixxxer** [comment]
> I think you are right with this... I'm a bit hesitant as this might be a breaking change...

I agree, it is used in many places, and in fact it does break `RedirectToIndexMiddleware`! I have to figure out how to fix it now...

**Comment by RicoSuter** [comment]
Indeed, if you find a way which just works i’m more then happy to merge it 🙂 but i doubt that this can be automatically fixed...

**Comment by ideafixxxer** [comment]
So far I worked around it the consumer code
``` C#
settings.TransformToExternalPath = (url, request) => url.EndsWith(".json", System.StringComparison.OrdinalIgnoreCase) 
  ? request.PathBase + url 
  : url;
```
Maybe `TransformToExternalPath` should be defined on the `SwaggerUi3Route` level?

**Comment by angusbreno** [comment]
![image](https://github.com/RicoSuter/NSwag/assets/5822893/f5ade73b-e0ee-4df4-b6e3-bf63372508ff)

I've to use the fix, because when using external specs the default imp adds '/' on the url start.
